frontend/packages/dev-console/src/components/helm/list/HelmReleaseList.tsx

```diff
@@ -13,22 +15,83 @@ import HelmReleaseListRow from './HelmReleaseListRow';
 import HelmReleaseListHeader from './HelmReleaseListHeader';
 import { HelmRelease } from '../helm-types';
 
+import './HelmReleaseList.scss';
+
 interface HelmReleaseListProps {
   namespace: string;
   secrets?: FirehoseResult;
 }
 
 const HelmReleaseList: React.FC<HelmReleaseListProps> = ({ namespace, secrets }) => {
-  const memoizedSecrets = useDeepCompareMemoize(secrets?.data);
+  const secretsCountRef = React.useRef<number>(0);
+  const newCount = secrets?.data?.length ?? 0;
+  const [releasesLoaded, setReleasesLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<string>();
+  const [releases, setReleases] = React.useState([]);
+
+  React.useEffect(() => {
+    setReleasesLoaded(false);
+    secretsCountRef.current = 0;
+  }, [namespace]);
 
   const getHelmReleases = (): Promise<HelmRelease[]> => {
-    return fetchHelmReleases(namespace);
+    if (!secrets || !secrets.loaded || secrets.loadError) {
+      return Promise.resolve([]);
+    }
+    if (newCount === 0) {
+      setLoadError(null);
+      setReleasesLoaded(true);
+      return Promise.resolve([]);
+    }
+
+    if (secretsCountRef.current === newCount) {
+      return Promise.resolve(releases);
+    }
+
+    secretsCountRef.current = newCount;
+    setReleasesLoaded(false);
+    return fetchHelmReleases(namespace)
+      .then((helmReleases) => {
+        setReleases(helmReleases);
+        setReleasesLoaded(true);
+        setLoadError(null);
+        return releases;
+      })
+      .catch((err) => {
+        setReleasesLoaded(true);
+        setLoadError(err.message || 'Unable to load Helm Releases');
+        return err;
+      });
+  };
+
+  if (secrets.loadError || loadError) {
```

Comment:
@jeff-phillips-18 I can still get the white screen because of this -

![Screenshot from 2020-05-14 21-26-42](https://user-images.githubusercontent.com/6041994/81957279-01ba4980-962a-11ea-8c27-b643f44a09fb.png)